frontend/src/GameRouter.tsx

```diff
@@ -100,7 +100,19 @@ const useSetupSockets = ({
   );
 
   // Round
-  const handlePunchlinesAdd = useCallback(addPunchlines, [addPunchlines]);
+  const handlePunchlinesAdd = useCallback(
+    (punchlines: string[]) =>
+      dispatchPunchlines({ type: PunchlinesAction.ADD, punchlines }),
+    [dispatchPunchlines]
+  );
+  const handlePunchlinesRemove = useCallback(
+    (punchlines: string[]) =>
+      dispatchPunchlines({
+        type: PunchlinesAction.REMOVE,
+        punchlines,
+      }),
+    [dispatchPunchlines]
+  );
```

Comment:
I know the intention was to clean up the contexts but it seems that complexity/boilerplate code has been increased

Reply:
Yeah there is some necessary increase in boilerplate `GameRouter` _for now_, although `GameRouter` itself is due for a refactor after all of this & I'd like to see it reduced too.

However overall complexity has definitely decreased. Lots of reasons for this, but the reducer pattern is definitely a nice way to test and the contexts are now hitting 100% coverage (basically 0% before). Also `useCrud()` will be able to be deleted once the punchlines and players contexts are refactored, which itself was not fully tested and included some functionality that was not implemented by the punchlines context.

Reply:
Right okay that sounds pretty good then